ui/src/common/ScrollUpButton.tsx

```diff
@@ -3,11 +3,36 @@ import KeyboardArrowUp from '@material-ui/icons/KeyboardArrowUp';
 import React, {Component} from 'react';
 
 class ScrollUpButton extends Component {
+    state = {
+        display: 'none',
+        opacity: 0,
+    };
+    componentDidMount() {
+        if (typeof window !== 'undefined') {
+            window.addEventListener('scroll', () => {
+                let currentScrollPos = window.pageYOffset;
+                if (currentScrollPos > 0) {
+                    this.setState({display: 'inherit'});
+                    this.setState({opacity: currentScrollPos / 500});
```

Comment:
Could you merge the 2 setStates into one? Same for the one below
```suggestion
                    this.setState({opacity: currentScrollPos / 500, opacity: Math.min(currentScrollPos / 500, 1)});
```
Also added Math.min that opacity doesn't exceed 1.